modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/js/components/ScriptElementAttributes.tsx

```diff
@@ -0,0 +1,279 @@
+			</ClayLayout.Col>
+
+			<ClayLayout.Col size={4}>
+				<div className="d-flex justify-content-between">
+					<label htmlFor="value">
+						{Liferay.Language.get('value')}
+					</label>
+
+					<div>
+						{index > 0 && (
+							<ClayButtonWithIcon
+								aria-label={Liferay.Language.get('remove')}
+								className="btn btn-primary btn-xs dm-field-repeatable-delete-button rounded-pill"
+								onClick={() => handleRemoveClick(index)}
+								symbol="hr"
+								title={Liferay.Language.get('remove')}
+								type="button"
+							/>
+						)}
+
+						<ClayButtonWithIcon
+							aria-label={Liferay.Language.get('add')}
+							className="btn btn-primary btn-xs dm-field-repeatable-add-button ml-1 rounded-pill"
+							onClick={() =>
+								attributeName
+									? handleAddClick(index)
+									: setAttrErrorMessage(
+											Liferay.Language.get(
+												'attribute-field-is-required'
+											)
+									  )
+							}
+							symbol="plus"
+							title={Liferay.Language.get('add')}
+							type="button"
+						/>
+					</div>
+				</div>
+
+				{attributeType !== 'Boolean' ? (
+					<ClayInput
+						id="value"
+						name={`${portletNamespace}value${index}`}
+						onChange={({target}) => {
+							const newValue = target.value;
+							setAttributeValue(newValue);
+						}}
+						type="text"
+					/>
+				) : (
+					<Picker
+						aria-labelledby="picker-label"
+						id="boolean"
+						items={booleanValue}
+						onSelectionChange={(value) => setAttributeValue(value)}
+						selectedKey={attributeValue.toString()}
+					>
+						{(item) => (
+							<Option key={item.value}>{item.label}</Option>
+						)}
+					</Picker>
+				)}
+			</ClayLayout.Col>
+		</ClayLayout.Row>
+	);
+};
+
+const ScriptElementAttributes = ({
+	attributesListUpdated: initialAttributesList,
+	portletNamespace,
+}) => {
+	const emptyRow = () => ({attribute: '', id: uuidv4(), type: '', value: ''});
+
+	const [attributesList, settAtributesList] = useState(
+		initialAttributesList && !!initialAttributesList.length
+			? [initialAttributesList.map((item) => ({...item, id: uuidv4()}))]
+			: [emptyRow()]
+	);
+
+	const addRow = (index) => {
+		const tempList = [...attributesList];
+		tempList.splice(index + 1, 0, emptyRow());
+		settAtributesList(tempList);
+	};
+
+	const removeRow = (index) => {
+		const tempList = [...attributesList];
+		tempList.splice(index, 1);
+		settAtributesList(tempList);
+	};
+
+	const updateAttributeList = useCallback((index, updatedValue) => {
+		const updatedString = {
+			attribute: updatedValue.attribute,
+			id: updatedValue.id,
+			type: updatedValue.type,
+			value: updatedValue.value,
+		};
+
+		settAtributesList((prevList) => {
+			const newList = [...prevList];
+			newList[index] = updatedString;
+
+			return newList;
+		});
+	}, []);
+
+	const filteredList = attributesList
+		.map(
+			({attribute, type, value}) =>
+				`${attribute}=${type === 'String' ? `"${value}"` : value}`
+		)
+		.join('\n');
```

Comment:
Let's make this a bit more explicit by converting this to an object first and then stringifying the object like this:
```
var toJSONObjectString = (attributes) => {
    const attributesObject = {};

    attributes.forEach(attribute => {
        attributesObject[attribute.name] = attribute.value;
    });

    return JSON.stringify(attributesObject);
}
```
We can also hoist this function outside of the component so it's not recreated during every render and just call it directly when we set the value of the input.